File: Domain/Author.cs
namespace Domain
{
    public class Author
    {
        public Guid AuthorId { get; set; }
        public string Forename { get; set; }
        public string Surename { get; set; }
        public string Bio { get; set; }
        public ICollection<BookAuthors> Books { get; } = new List<BookAuthors>();
    }
}

File: client-app/src/app/models/author.ts
export interface Author {
    authorId: string;
    forename: string;
    surename: string;
    bio: string;
}

export class Author implements Author {
    constructor(init?: AuthorFormValues) {
        Object.assign(this, init);
    }
}

export class AuthorFormValues {
    authorId?: string = undefined;
    forename: string = "";
    surename: string = "";
    bio: string = "";

    constructor(author?: AuthorFormValues) {
        if(author) {
            this.authorId = author.authorId;
            this.forename = author.forename;
            this.surename = author.surename;
            this.bio = author.bio;
        }
    }
}